mmcv/ops/csrc/pytorch/cuda/vector_pool.cu

```diff
@@ -0,0 +1,137 @@
+    const Tensor new_xyz_grid_centers_tensor, Tensor new_xyz_grid_idxs_tensor,
+    Tensor new_xyz_grid_dist2_tensor, Tensor stack_neighbor_idxs_tensor,
+    Tensor start_len_tensor, const int M, const int num_total_grids) {
+  at::cuda::CUDAGuard device_guard(support_xyz_tensor.device());
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  // blockIdx.x(col), blockIdx.y(row)
+  dim3 blocks(DIVUP(M, THREADS_PER_BLOCK), num_total_grids);
+  dim3 threads(THREADS_PER_BLOCK);
+
+  query_three_nn_by_stacked_local_idxs_cuda_kernel<<<blocks, threads>>>(
+      support_xyz_tensor.data_ptr<float>(), new_xyz_tensor.data_ptr<float>(),
+      new_xyz_grid_centers_tensor.data_ptr<float>(),
+      new_xyz_grid_idxs_tensor.data_ptr<int>(),
+      new_xyz_grid_dist2_tensor.data_ptr<float>(),
+      stack_neighbor_idxs_tensor.data_ptr<int>(),
+      start_len_tensor.data_ptr<int>(), M, num_total_grids);
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+int StackVectorPoolForwardCUDAKernelLauncher(
+    const Tensor support_xyz_tensor, const Tensor xyz_batch_cnt_tensor,
+    const Tensor support_features_tensor, const Tensor new_xyz_tensor,
+    const Tensor new_xyz_batch_cnt_tensor, Tensor new_features_tensor,
+    Tensor new_local_xyz_tensor, Tensor point_cnt_of_grid_tensor,
+    Tensor grouped_idxs_tensor, const int num_grid_x, const int num_grid_y,
+    const int num_grid_z, const float max_neighbour_distance, const int use_xyz,
+    const int num_max_sum_points, const int nsample, const int neighbor_type,
+    const int pooling_type) {
+  at::cuda::CUDAGuard device_guard(support_xyz_tensor.device());
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  const float *support_xyz = support_xyz_tensor.data<float>();
+  const float *support_features = support_features_tensor.data<float>();
+  const int *xyz_batch_cnt = xyz_batch_cnt_tensor.data<int>();
+  const float *new_xyz = new_xyz_tensor.data<float>();
+  const int *new_xyz_batch_cnt = new_xyz_batch_cnt_tensor.data<int>();
+  float *new_features = new_features_tensor.data<float>();
+  float *new_local_xyz = new_local_xyz_tensor.data<float>();
+  int *point_cnt_of_grid = point_cnt_of_grid_tensor.data<int>();
+  int *grouped_idxs = grouped_idxs_tensor.data<int>();
+  int N = support_xyz_tensor.size(0);
+  int batch_size = xyz_batch_cnt_tensor.size(0);
+  int M = new_xyz_tensor.size(0);
+  int num_c_out = new_features_tensor.size(1);
+  int num_c_in = support_features_tensor.size(1);
+  int num_total_grids = point_cnt_of_grid_tensor.size(1);
+
+  int num_c_each_grid = num_c_out / num_total_grids;
+  float grid_size_x = max_neighbour_distance * 2 / num_grid_x;
+  float grid_size_y = max_neighbour_distance * 2 / num_grid_y;
+  float grid_size_z = max_neighbour_distance * 2 / num_grid_z;
+  dim3 blocks(DIVUP(M, THREADS_PER_BLOCK));  // blockIdx.x(col), blockIdx.y(row)
+  dim3 threads(THREADS_PER_BLOCK);
+  int cum_sum = 0;
+  int *p_cum_sum;
+  cudaMalloc((void **)&p_cum_sum, sizeof(int));
```

Comment:
Do not use cuda malloc in the implementation.
Why not return a tensor with single element?